### PR TITLE
(#13) Redirect to Pretty URL

### LINKED ISSUE #13: Story: Redirect to Pretty URL
## As the website, I should redirect CDR IDs in URLs to its counterpart pretty URL for better findability.

> **ESTIMATE** XXX

### Acceptance Criteria

```gherkin
Scenario: Redirect from CDR ID
  Given user enters /def/42265 as the URL
  Then the system redirects user to /def/trastuzumab
  And the system appends ?redirect=true to the URL
Examples:
| CDR ID | URL Redirect|
| 43234 | bevacizumab |
| 484642 | jin-fu-kang |
| 269682 | juvin |
```

### Resources:
* Feature: JIRA Ticket #
* Mockups: *Here goes a URL to or the name of the mockup(s) in inVision*
* Web Page: *Here goes a URL to an existing web page*

### Notes

*Some complementary notes if necessary:*

* > Here goes a quote from an email
* Here goes whatever useful information can exist…

---

### Prerequisites
- *required dependency for identifying the problem, scope and then solution for this issue*
- [ ] Short Spike Description - Issue #9999

### Tasks
- [ ] Short Task Description - Issue #9999
- [ ] Short Task Description - Issue #9999

### Technical Notes
- More notes, or technical nuances/direction

### PR DESCRIPTION
**Pull request by KateMashkinaNIH**
- Updated definition component to redirect to pretty url when ID is provided
- Added unit tests for prettyUrl redirect in definition
- Added integration test

Closes #13

### CODE REVIEW

File: src/views/Definition/__tests__/Definition.PrettyURLRedirect.test.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+	const expectedLocationObject = {
+		pathname: '/def/xenogeneic-tyrosinase-dna-vaccine',
+		search: '?redirect=true',
+		hash: '',
+		state: null,
+		key: expect.any(String),
+	};
+	expect(location).toMatchObject(expectedLocationObject);
+});
```
Comment (arcepaul) [inline src/views/Definition/__tests__/Definition.PrettyURLRedirect.test.js:92]: this mark tells me that you need to run format on this file

Comment (kate-mashkina) [reply src/views/Definition/__tests__/Definition.PrettyURLRedirect.test.js]: Formatted!

File: cypress/integration/PrettyUrlRedirect.feature [hunk 2]
```diff
@@ -0,0 +1,7 @@
+Feature: As the website, I should redirect CDR IDs in URLs to its counterpart pretty URL for better findability
+
+
+    Scenario: Redirect from CDR ID
+        Given the user navigates to "/def/750633"
+        Then the system redirects user to "/def/acenocoumarol"
+        And the system appends "?redirect=true" to the URL
```
Comment (arcepaul) [inline cypress/integration/PrettyUrlRedirect.feature:7]: there's that symbol again!

Comment (kate-mashkina) [reply cypress/integration/PrettyUrlRedirect.feature:7]: Weird - I ran **npm run format** and this was not in the list of formatted files

Comment (kate-mashkina) [reply cypress/integration/PrettyUrlRedirect.feature]: Removed

**Review by seyilonge-nci — APPROVED** [review]

**Review by karlikpj — APPROVED** [review]

**Comment by kate-mashkina** [comment]
@VictoriaSunNIH @mworrest  Please review the PR https://react-app-dev.cancer.gov/drug-dictionary-app/pr-30/def/750633

**Comment by arcepaul** [comment]
For testers, 
find the cdrid for the drug you wish to use and replace it in the url.  It should redirect to the prettyUrl name

**Aceon** (660818)
https://react-app-dev.cancer.gov/drug-dictionary-app/pr-30/def/660818

**Borocarpin** (306451)
https://react-app-dev.cancer.gov/drug-dictionary-app/pr-30/def/306451

**Comment by blilianyu** [comment]
This is working as expected! @VictoriaSunNIH @mworrest

**Review by arcepaul — APPROVED** [review]

**Pull request merged**